src/server/admin_realm.cpp

```diff
@@ -75,10 +68,13 @@ void AdminRealmListener::start()
         // If we've finished downloading the Realm, just re-report all the files listed in it
         auto& group = realm->read_group();
         auto& table = *ObjectStore::table_for_object_type(group, "RealmFile");
-        size_t path_col_ndx = table.get_column_index("path");
+        auto path_col_key = table.get_column_key("path");
 
-        for (size_t i = 0, size = table.size(); i < size; ++i)
-            register_realm(id_for_row(group, table[i]), table.get_string(path_col_ndx, i));
+        for (auto it = table.begin(); it != table.end(); ++it) {
+            auto id = table.get_object_id(it->get_key());
+            auto obj = table.get_object(it->get_key());
+            register_realm(id,  obj.get<StringData>(path_col_key));
+        }
```

Comment:
Be aware that `it` is a pointer to current Obj, so no need to get the object again.
```
for (auto& o : table)
    register_realm(table.get_object_id(o.get_key()), o.get<String>(path_col_ndx));
```
is shorter.